### DISCUSSION TITLE
Reaction button  hidden on mobile

### DISCUSSION THREAD

**Discussion by Slion** [community/community#12781] [Mobile]
Though is shows for my own comments but not on other people's comments. Also if a comment already has some reaction then you can still use the reaction button below the comment. However if a comment is not yours and does not have a reaction already then you can't react to it because the reaction button at the top next to the menu button is not shown.

I feel like this is a regression that was introduced recently.

**Comment by michaelblyons** [comment]
I sometimes see a reaction button in the header of a comment even when there are no reactions on it, but it is rare, and I have no idea what the pattern for that might be. (Maybe someone reacted and then un-reacted?)

But this is indeed annoying. I would love for this to be fixed.

**Reply by genofire** [reply]
I just see it, if i am in Desktop vor Desktop-mode in Browser.

**Reply by michaelblyons** [reply]
I now see it every time, but clicking on it zooms out the whole page so the emoji can load to the right of the content area.

**Reply by genofire** [reply]
You mean this way? Yes, me too.

![Screenshot_20220513-003116_Fennec_1](https://user-images.githubusercontent.com/6905586/168178317-e2512771-5a9c-4578-9cd3-a2dfae3ffb21.png)

**Comment by beerisgood** [comment]
can confirm this on Safari under iOS 15.4.1.

Screenshot:
![image](https://user-images.githubusercontent.com/9625159/168065370-2778133d-5539-482e-8bb1-528db2659852.png)

**Reply by thw0rted** [reply]
Seeing the same thing on Chrome / Android. I'm surprised this isn't getting more traction - I feel like it's a regression?

ETA: ironically the reaction button shows for me here (in a discussion), next to the overflow (ellipsis) menu, but not in issues.

ETA again: mystery solved, it's definitely buggy CSS. Rotate your device and the button will appear - it just needs enough space. It must overflow and be hidden in portrait mode.

**Reply by michaelblyons** [reply]
Buggy CSS!? Is it like "might show, depending on Poster's name/fuzzy-time"? Because that would be nuts.

**Reply by reduckted** [reply]
> Rotate your device and the button will appear 

That doesn't fix it for me.

**Reply by michaelblyons** [reply]
Landscape mode also does not affect it for me. I still see the reactions some places (where one already exists?), but not others.

**Reply by thw0rted** [reply]
Are you sure you're looking in the right place?  Compare portrait:

<img src="https://user-images.githubusercontent.com/943160/184815688-b036f802-d159-425b-8583-4cba9be3f732.png" height="400px">

to landscape:

<img src="https://user-images.githubusercontent.com/943160/184815767-0f2ade9f-baf7-4f6f-a078-8b6b71104463.png">

**Reply by reduckted** [reply]
This is what it looks like for me at [this comment](https://github.com/microsoft/vsmarketplace/issues/292#issuecomment-1211401806):

**Portrait**
![Screenshot_20220816-192645~2](https://user-images.githubusercontent.com/10321525/184847322-280ba12d-7ab0-4680-9ad2-d5b5adff8288.png)

**Landscape**
![Screenshot_20220816-192609~4](https://user-images.githubusercontent.com/10321525/184847412-55190b14-6092-4248-88b1-05172dae9e7a.png)

**Reply by thw0rted** [reply]
If you keep scrolling for a few comments to one that doesn't have a "member" or "author" tag (i.e. just the overflow menu), is that also missing the reaction button?  I'm wondering if, say, portrait has room for one button and landscape has room for two.  (I can't get to my phone at the moment, or I'd try it myself.)

**Reply by reduckted** [reply]
I had the same thought, but no, even if there are no "member" tags the reaction button doesn't show up.

You're right about the CSS, though. I just had a look on desktop, and when resizing the window to 767px wide or less, the reaction button is assigned the `d-none` class, which equates to `display: none !important;`.

I guess it may depend on how tall your phone is as to whether it reappears when rotating to landscape mode. Actually, on your landscape screenshot it looks like the right-hand side menu (with the _Subscribe_ button) is visible, which only appeared for me when the screen width was 768px or higher.

**Reply by alialhammad1992** [reply]
The problem is supposed to be solved from the ground up by supporting the GithHub application

**Comment by beerisgood** [comment]
Looks like the problem is back on Safari. 
no button available

**Comment by Slion** [comment]
Was it ever really fixed? I think they put the button back but the reaction selector pop-up would break the page layout and cause a zoom out of the page until you pick one reaction.

**Comment by BDisp** [comment]
To aggravate the situation this problem is also already verified in any browser executed from a laptop. The popup with the buttons appears outside the screen area resulting in some hidden buttons.
**Edit:**
I only refer to clicking the button located at the bottom of a comment of a pull request, after there has already been a reaction.

**Reply by alialhammad1992** [reply]
The problem is supposed to be solved from the ground up by supporting the GithHub application

**Comment by tmccombs** [comment]
This is still an issue for me in Firefox for Android.

**Comment by mag122ed** [comment]
31387933

**Comment by tmccombs** [comment]
Interestingly, the emoji button does show up in discussions, and if I am in landscape mode.

**Comment by beerisgood** [comment]
Today I got this broken popup in iOS Safari:
![5C20EDEE-26D2-466D-B647-C273A4684BA4](https://user-images.githubusercontent.com/9625159/201712406-0c1e7398-3e26-4928-9297-9debb43d1a98.jpeg)

**Comment by ProtoDot** [comment]
I would react to this since I don't have anything to add but I can't. 😡

**Comment by diegobotelhohm** [comment]
Você mora no meu coração. Gratidão por tudo Enviado do Email para Windows De: ProtoDotEnviado:segunda-feira, 5 de dezembro de 2022 21:28Para: community/communityCc:SubscribedAssunto: Re: [community/community] Reaction button hidden on mobile (Discussion #12781) I would react to this since I don't have anything to add but I can't. 😡—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

**Comment by diegobotelhohm** [comment]
Voce Faz Por Merecer, Basta Querer, Estar Dispoto a se reconhecer, Fazer acontecer, Fazer por você, Voce Pode TUDOOOO, quem faz acontecer e você mesmo, Seja mais leve com você mesmo. E se cobre menos, errar e humano, somos cheios de defeito, Estamos aqui pra aprendermos e sermos melhores.  Enviado do Email para Windows De: ProtoDotEnviado:segunda-feira, 5 de dezembro de 2022 21:28Para: community/communityCc:SubscribedAssunto: Re: [community/community] Reaction button hidden on mobile (Discussion #12781) I would react to this since I don't have anything to add but I can't. 😡—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

**Comment by telF4** [comment]
may be,that is future, laughing.

**Comment by xenago** [comment]
This is still an issue on Chromium based browsers on Android like Samsung Internet

**Reply by marc-guenther** [reply]
Still no button on Safari on my iPhone. Neither in portrait nor landscape mode. There is more than enough space available in the upper right. Also nothing in the ... context menu.

There IS a button, if somebody else already reacted to the comment, though. That one is in the lower left, though.

Clearly a company such as Github should be able to place a simple button on a page, without breaking the layout? How hard can it be?

**Comment by jefro108** [comment]
It appears to be an issue not in the ‘Discussions’ but in the ‘Issues’ section which can’t be seen on this repository. I’m on iPhone iOS 16.2 on Chrome. There are no reaction buttons visible unless someone has already used one in that particular comment. 

In ‘Discussions’ the reaction button is always next to the ‘upvote’ button under the comment, but in ‘Issues’ (where there’s no upvote button) it’s in the comment header when there is no previous reaction and doesn’t appear on mobile. 

However the reaction button appears in the header of a reply in a ‘Discussion’ (which we can see here) when it’s nested but replies aren’t nested in ‘Issues’.

It would solve the issue if in ‘Issues’ the reactions button was always placed below the comment - which is a more intuitive position anyway!

**Reply by BDisp** [reply]
@jefro108 why don't you open an issue based on your comment? Copy link and past on a new issue. It seems that the only place to open an issue is in this repo https://github.com/community/OSPO/issues.

**Comment by Kaminem202** [comment]
> Though is shows for my own comments but not on other people's comments. Also if a comment already has some reaction then you can still use the reaction button below the comment. However if a comment is not yours and does not have a reaction already then you can't react to it because the reaction button at the top next to the menu button is not shown.
> 
> I feel like this is a regression that was introduced recently.

**Comment by ckujau** [comment]
This is only the community ~buffer~ discussion forum, does anyone know if a GitHub bug exists for this problem?

**Comment by tan-wei-xin-alez** [comment]
Still happening on iOS Safari 15.1 (tested on comments on PR in private repo with both portrait and landscape mode), doesn’t seem to have an open GitHub issue as of now

**Comment by ADTC** [comment]
Using Android, and I find this so annoying. I have to enable "Desktop mode" to add the **first** reaction to a comment.

This affects the Issues section of a repository btw, not Discussions.

**Reply by BDisp** [reply]
> Using Android, and I find this so annoying. I have to enable "Desktop mode" to add the **first** reaction to a comment.
> 
Where do you enable the "Desktop mode" using Android?

**Reply by ADTC** [reply]
In Chrome browser in Android, you can open the menu and there will be a checkbox called "Desktop site".

I think other mobile browsers have a similar feature.

**Reply by BDisp** [reply]
Thanks. I found it. I never realized that option was there :-)

**Reply by ADTC** [reply]
There's a lot you can discover when you tap and swipe on everything you see on your phone screen ;)

**Comment by Djfe** [comment]
finally, we have the option now

**Comment by Slion** [comment]
Only took them about a year to fix it. To be honest I'm fully expecting them to break it again sooner or later 🥳

**Comment by rooby** [comment]
This seems to be broken again. Using Chrome on iOS.
EDIT: Seems to be working on issues but broken on PRs.

**Comment by Djfe** [comment]
Working just fine for me on issues as well as prs.
Firefox for Android (117.1.10)

I didn't notice anything broken lately either

**Comment by rooby** [comment]
Oh, the particular PR I'm looking at won't let me emote even on desktop. Seems to be something about that specific PR.
Apologies for the noise.
